test/TesterInternal/ReentrancyTests.cs

```diff
@@ -342,6 +482,55 @@ public void FanOut_AC_Stress_NonReentrant()
             Assert.True(elapsed < MaxStressExecutionTime, $"Stress test execution took too long: {elapsed}");
         }
     }
+
+    public class MessageInterleavesPredicateBootstrapProvider : IBootstrapProvider
+    {
+        public string Name { get; private set; }
+
+        public Task Init(string name, IProviderRuntime providerRuntime, IProviderConfiguration config)
+        {
+            Name = name;
+            InstallPredicate();
+            return TaskDone.Done;
+        }
+
+        void InstallPredicate()
+        {
+            GrainTypeData grainType;
+            GrainTypeManager.Instance.TryGetData(typeof(NonReentrantGrainWithMessageInterleavePredicate).FullName, out grainType);
+            grainType.SetMessageAllowedToInterleavePredicate(req =>
```

Comment:
I'm not sure. From the discussion above it was not clear what was the final decision. Should be a class-level attribute for this to work with streams. If that is a general consensus - I'll take this route